src/Rule/Nested.php

```diff
@@ -36,9 +37,86 @@
 use function sprintf;
 
 /**
- * Can be used for validation of nested structures.
+ * Used to define rules for validation of nested structures:
+ *
+ * - For one-to-one relation, using `Nested` rule is enough.
+ * - One-to-many and many-to-many relations require pairing with {@see Each} rule.
+ *
+ * An example with blog post:
+ *
+ * ```php
+ * $rule = new Nested([
+ *     'title' => [new HasLength(max: 255)],
+ *     // One-to-one relation
+ *     'author' => new Nested([
+ *         'name' => [new HasLength(min: 1)],
+ *     ]),
+ *     // One-to-many relation
+ *     'files' => new Each([
+ *          new Nested([
+ *             'url' => [new Url()],
+ *         ]),
+ *     ]),
+ * ]);
+ * ```
+ *
+ * There is an alternative way to write this using dot notation and shortcuts:
+ *
+ * ```php
+ * $rule = new Nested([
+ *     'title' => [new HasLength(max: 255)],
+ *     'author.name' => [new HasLength(min: 1)],
+ *     'files.*.url' => [new Url()],
+ * ]);
+ * ```
+ *
+ * Note that the maximum nesting level is 2, a deeper one requires wrapping with another `Nested` instance or using
```

Comment:
Doc is fine but the limitation itself isn't great. Is there an issue to improve it?

Reply:
I was going to create one. :ok_hand:

Reply:
#508